backend-services/voting-app/src/main/java/org/cardano/foundation/voting/client/KeriVerificationClient.java

```diff
@@ -145,8 +148,91 @@ public Either<Problem, String> getOOBI(String oobi, Integer maxAttempts) {
         return Either.left(Problem.builder()
                 .withTitle("OOBI_NOT_FOUND")
                 .withDetail("The OOBI was not found after " + attempts + " attempts.")
-                .withStatus(new HttpStatusAdapter(BAD_REQUEST))
+                .withStatus(new HttpStatusAdapter(NOT_FOUND))
                 .build());
     }
 
+    public Either<Problem, Boolean> updateAndVerifyKeyState(String aid, Integer maxAttempts) {
+        val updateUrl = String.format("%s/keystate", keriVerifierBaseUrl);
+        val verifyUrl = String.format("%s/keystate/%s", keriVerifierBaseUrl, aid);
+
+        val headers = new HttpHeaders();
+        headers.add("Content-Type", "application/json");
+
+        val requestBody = new HashMap<String, String>();
+        requestBody.put("pre", aid);
+
+        val entity = new HttpEntity<Map<String, String>>(requestBody, headers);
+
+        // Attempt to update the key state
+        try {
+            val response = restTemplate.exchange(updateUrl, POST, entity, String.class);
+            if (response.getStatusCode().is2xxSuccessful()) {
+                log.info("Key state updated successfully for aid: {}", aid);
```

Comment:
`updateUrl` triggers an update and returns a 202 accepted. If that fails, then we need to return a 500 or something to the user.

Once 202 on `updateUrl`, we poll `verifyUrl` for completion. Currently we aren't waiting which is why the signature verification fails after rotating keys